Implement First-Party Mode Setup Success Subtle Notification Component

techanvil opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #9648 

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

nfmohit commented on assets/js/components/notifications/FirstPartyModeSetupSuccessSubtleNotification.stories.js:
```diff
@@ -0,0 +1,58 @@
+
+export default {
+	title: 'Modules/FirstPartyMode/Dashboard/FirstPartyModeSetupSuccessSubtleNotification',
+	decorators: [
+		( Story ) => {
+			const setupRegistry = ( registry ) => {
+				registry
+					.dispatch( CORE_SITE )
+					.receiveGetFirstPartyModeSettings( {
+						isEnabled: true,
+					} );
+			};
+
+			return (
+				<WithRegistrySetup func={ setupRegistry }>
+					<Story />
+				</WithRegistrySetup>
+			);
+		},
+	],
```
I think we can get away with not mutating the registry here, as the component should render regardless. What do you think?

```suggestion
```

techanvil replied:
You're quite right, this is unnecessary and I've removed it. Thanks for the spot!

nfmohit commented on assets/js/components/notifications/FirstPartyModeSetupSuccessSubtleNotification.test.js:
```diff
@@ -0,0 +1,83 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createTestRegistry, render } from '../../../../tests/js/test-utils';
```
While we're at it, I think we can keep up the ritual and add a comment about the dependency type.

```suggestion

/**
 * Internal dependencies
 */
import { createTestRegistry, render } from '../../../../tests/js/test-utils';
```

techanvil replied:
Absolutely, thanks for spotting this one! Updated.

nfmohit commented on assets/js/googlesitekit/notifications/register-defaults.js:
```diff
@@ -479,6 +483,18 @@ export const DEFAULT_NOTIFICATIONS = {
 	},
+	'setup-success-notification-fpm': {
+		Component: FirstPartyModeSetupSuccessSubtleNotification,
+		priority: 310,
```
Should we consider prioritising this notification as it is "on-demand"/appears upon action? The IB mentions setting it to a 10, but I'm interested to hear your thoughts, thank you!

techanvil replied:
Hmm yes, I thought I'd added this as a 10, maybe an overenthusiastic AI changed it for me :thinking: 

Anyhow good shout, I've updated this to follow the IB, cheers.

nfmohit commented on assets/js/components/notifications/FirstPartyModeSetupBanner.js:
```diff
@@ -60,10 +67,20 @@ export default function FirstPartyModeSetupBanner( { id, Notification } ) {
+
 	const onCTAClick = () => {
 		setFirstPartyModeEnabled( true );
 		saveFirstPartyModeSettings();
```
Should we consider (a)waiting for this fetch to complete, and only proceed if there is no error?

techanvil replied:
Thanks for raising that - I was meaning to create an issue to add error handling for this component. I've done so now: https://github.com/google/site-kit-wp/issues/9846

nfmohit commented on assets/js/components/notifications/FirstPartyModeSetupSuccessSubtleNotification.js:
This component looks good, but has a slight discrepancy with the Figma designs. I personally feel the design should be updated as it is not consistent with the existing `SubtleNotification` components in SK.

1. The vertical padding is 18px, while in the designs, it is 14px.
2. The gap between the flex elements is 16px, while in the designs, it is 30px.

It is worth noting that the first discrepancy regarding the vertical padding also exists in the `FirstPartyModeToggle` component's warning subtle notification, which I assumed was a one time exception and [applied that using its stylesheet](https://github.com/google/site-kit-wp/pull/9787/files#diff-1d512a77e0efeeb5013160d6ceb75b57a7b989a4fb7f48af51b4acf1088aed08R31).

techanvil replied:
Thanks for spotting that - you're quite right, we should have a set of standard components with consistent designs and this one should have the same layout as the other instances of it.

I've [raised that](https://www.figma.com/design/udlPW35VZ0EDnSbRxYOGMj?node-id=56-2795&m=dev#1056620654) with @sigal-teller, in the meantime let's proceed as implemented. I've also made a note in the QAB about it.

nfmohit reviewed (changes requested):
Brilliant work here, thank you @techanvil ! I've addd a few mostly minor comments and questions, please let me know what you think, thank you!

techanvil commented on assets/js/components/notifications/FirstPartyModeSetupSuccessSubtleNotification.stories.js:
```diff
@@ -0,0 +1,58 @@
+/**
+ * AudienceSegmentationSetupSuccessSubtleNotification Component Stories.
```
Good spot, thanks! Updated.

techanvil commented:
Thanks for your review @nfmohit! I've addressed your feedback. Please note the JS test suite has been a bit flaky so I've sent it back over with an unrelated failing test. Looks like we need a GH issue to tackle these flakes, I've made a note to add one as time allows (or please feel free to do so if you have capacity at the moment).

nfmohit reviewed (approved):
LGTM ✅ 

Thank you for addressing the feedback, @techanvil! I reran the failing tests in CI and they all passed. I'll open an issue if I encounter this again, unless you do so before I can, thanks!
